Add community guidelines (#6528)
gammapy/gammapy: docs/development/guidelines.rst

.. _dev_guidelines:

====================
Community Guidelines
====================

This section describes a list of requirements and guidelines that should be followed by
any contributor to Gammapy.

* If you are completely new to Gammapy, both as a user and contributor, we insist that
  you get in `contact <https://gammapy.org/contact.html>`__ with us early via GitHub,
  Slack or email before beginning your journey
* First, you should comment on an issue (ideally labeled package-novice) to express
  your interest, and propose an approach there.
* Opening a PR without prior discussion is not accepted.
* Please follow the AI guidelines outlined in detail below

Failure to comply with these guidelines will result in a warning, your PRs not being
merged... etc.
For further details about contributing to gammapy, please familiarise yourself with
the :ref:`Developer Guide <dev_intro>` to make sure PRs comply with our coding standards
and style.

AI Guidelines
=============

> insert guidelines here

Author: Yes, i don't think we need to include prompts

  your interest, and propose your intended approach there.
* Opening a pull request (PR) without prior discussion is generally not accepted.
* Please follow the authorship policy outlined :ref:`here <authorship-policy>`.
* While we welcome new contributors, contributions should be meaningful. PRs consisting only of very
  minor documentation changes are discouraged.
* Please follow the AI guidelines outlined in detail below.

If these guidelines are not followed, developers may issue you a warning or your PR may not be accepted.

For further details about contributing to Gammapy, please familiarise yourself with
the :ref:`Developer Guide <dev_intro>` to make sure PRs comply with our coding standards
and style.

AI Guidelines
=============

We recognise that artificial intelligence (AI), in particular Large language models (LLMs),
is a modern tool in many aspects of the world and acknowledge its wide use within software development.
We list below a number of guidelines for utilising AI:

* Any usage of AI must be clearly detailed in the PR
* The responsibility of contributing non copyrighted code rests solely on the author
* The author needs to be able to explain the all commits

> * Avoid directly copy pasting AI generated content for either the PR text or content

Reviewer: Then maybe insist on understanding:
- Only submit code or text that you have reviewed, tested and understood (in particular avoid copy pasting AI generated content)

.. _dev_guidelines:

====================
Community Guidelines
====================

This section describes a list of requirements and guidelines that should be followed by
any contributor to Gammapy. We warmly welcome new contributors to Gammapy.

* If you are new to Gammapy, both as a user and contributor, we strongly encourage you
  to get in `contact <https://gammapy.org/contact.html>`__ with us early via GitHub,
  Slack or email before beginning your journey. See :ref:`contact-early` for further detail.

> * Before beginning work, you should comment on an issue (ideally labeled ``package-novice``) to express

Reviewer 2: Before starting a PR (instead of beginning work, which is more general)?

> * Before beginning work, you should comment on an issue (ideally labeled ``package-novice``) to express

Reviewer 2: Or open a new issue if they have something specific in mind

  your interest, and propose your intended approach there.
* Opening a pull request (PR) without prior discussion is generally not accepted.

> * Please follow the authorship policy outlined :ref:`here <authorship-policy>`.

Reviewer 2: We can make this statement stronger to emphasise strongly on following the DCO to avoid maintenance effort during releases

Author: What do you suggest?

Reviewer 2: All authors must follow the DCO and ensure their git account is correctly configured
- in bold :)

* While we welcome new contributors, contributions should be meaningful. PRs consisting only of very
  minor documentation changes are discouraged.
* Please follow the AI guidelines outlined in detail below.

> If these guidelines are not followed, developers may issue you a warning or your PR may not be accepted.

Reviewer 2: A threat of a warning does not sound very useful.
It scares well intentioned contributors, and does nothing to discourage spammers.

Maybe say "PRs not adhering to these guidelines will not be accepted"

Author: I am happy to change it, but a warning is not very strong really just to say 'please dont do that'

For further details about contributing to Gammapy, please familiarise yourself with
the :ref:`Developer Guide <dev_intro>` to make sure PRs comply with our coding standards
and style.

AI Guidelines
=============

We recognise that artificial intelligence (AI), in particular Large language models (LLMs),
is a modern tool in many aspects of the world and acknowledge its wide use within software development.
We list below a number of guidelines for utilising AI:

> * Any usage of AI must be clearly stated in the PR

Reviewer: I think we can even state the "Ell PR must contain an AI usage disclosure"
and we can add it in the PR template.

Author: Yes good idea, I will add it now!

Also should I add something like `Do not use AI to communicate with us.`?

Reviewer: Well I suppose that translation is fine, no? we just don't want to talk to a bot

Author: Yes, but one can you translate apps for that. In the end, our current issue is that majority of the 'new comers' are just bots. 
Maybe:
`You should avoid using AI to communicate with us, so we can ensure you are not an automated tool (i.e. a bot)`

> * The responsibility of contributing non copyrighted code rests solely on the author

Reviewer 3: This is tricky, because we have no way to blame afterward an author. This is **our** responsability, as core team because we accept contribution
We should rather stated that codes that does not follow our licence is forbiden. But, are some ouputs of AI non free? If no, this line is useful. If yes, one should state strongly that only BSD-3 clause code is accepted.

Reviewer 2: DIsucssion in dev call: Rephrase to say "Ensure it is compatible with the Gammapy Licensing rules [BSD 3]"

Author: Okay but this is in the line below?

Author: So rather remove this all together?

Reviewer 2: @Astro-Kirsty we can say that the authors should ensure that all contributions are compatible with our standard license (which includes copyright etc), instead of explicitly saying that the responsibility rests solely with the author (because in the end, it's our responsibility that we don't merge junk)

* The author needs to be able to explain the all commits
* Avoid directly copy pasting AI generated content for either the PR text or content
* Purely automated PRs will be closed without discussion

> It is important to note, the Gammapy package maintainers reserve the right to close any PR not

Reviewer: Same as above: Any PR not adhering to the requirements will be closed

adhering to the requirements

In summary, we ask that you keep the two following key points in mind:

> * Never let an LLM speak for you.

Reviewer 2: I like this :-)

.. _dev_guidelines:

====================
Community Guidelines
====================

This section describes a list of requirements and guidelines that should be followed
by any contributor to Gammapy. We warmly welcome new contributors to Gammapy.

> * If you are new to Gammapy please read the :ref:<new-contributor> steps in addition

Reviewer 3: ```suggestion
* Any Gammapy activity should respect the `Code of Conduct <https://gammapy.org/CoC.html>`__ of the projet.
* If you are new to Gammapy please read the :ref:<new-contributor> steps in addition
```

  the ones outlined here.
* Before beginning work on a pull request (PR), you should comment on an issue
  (ideally labeled ``package-novice``) to express your interest, and propose
  your intended approach there. If you have a specific idea (e.g. you notice a bug
  or want to propose a new feature) you should open an issue before a PR.
* Opening a PR without prior discussion is generally not accepted.
* All authors must follow the the authorship policy outlined :ref:`here <authorship-policy>`
  and ensure their git account is correctly configured.

> * While we welcome new contributors, contributions should be meaningful. PRs

Reviewer 3: ```suggestion
* Anyone should read and promote the Gammapy `citation scheme <https://gammapy.org/acknowledging.html>`__.
* While we welcome new contributors, contributions should be meaningful. PRs
```

  your intended approach there. If you have a specific idea in mind you should
  open an issue before a PR.
* Opening a PR without prior discussion is generally not accepted.
* Please follow the authorship policy outlined :ref:`here <authorship-policy>`.
* While we welcome new contributors, contributions should be meaningful. PRs
  consisting only of very minor documentation changes are discouraged.
* Please follow the AI guidelines outlined in detail below.

Any PRs not adhering to these guidelines will not be accepted.

For further details about contributing to Gammapy, please familiarise yourself with
the :ref:`Developer Guide <dev_intro>` to make sure PRs comply with our coding
standards and style.

AI Guidelines
=============

We recognise that artificial intelligence (AI), in particular Large language models
(LLMs), is a modern tool in many aspects of the world and acknowledge its wide use
within software development. We list below a number of guidelines for utilising AI:

* All PRs must include a clear disclosure of any AI usage
* You should avoid using AI to communicate with us, so we can ensure you are not an

>   automated tool (i.e. a bot)

Reviewer: I think this is not really needed given the ones below

Author: So you propose to remove this all together?

Reviewer 2: +1 to removing this

> * The responsibility of contributing non copyrighted code rests solely on the author

Reviewer: Maybe mention the DCO and link to it here.

Reviewer: Maybe: Contributions must comply with Gammapy's license and Developer cetrficate of origin (DCO)

> * The author needs to be able to explain the all commits

Reviewer: ```suggestion
* The author needs to be able to explain the all the changes introduced
```

.. _dev_guidelines:

====================
Community Guidelines
====================

This section describes a list of requirements and guidelines that should be followed
by any contributor to Gammapy. We warmly welcome new contributors to Gammapy.

> * If you are new to Gammapy, both as a user and contributor, we strongly encourage

Reviewer 2: This I would put in the "New contributors" section

  you to get in `contact <https://gammapy.org/contact.html>`__ with us early via
  GitHub,  Slack or email before beginning your journey. See :ref:`contact-early`
  for further detail.
* Before beginning work on a pull request (PR), you should comment on an issue
  (ideally labeled ``package-novice``) to express your interest, and propose

>   your intended approach there. If you have a specific idea in mind you should

Reviewer 2: ```suggestion
  your intended approach there. If you have a specific idea (eg: you notice a bug or want to propose a new feature) in mind you should
```
